app/code/core/Mage/Customer/controllers/AccountController.php

```diff
@@ -715,6 +715,13 @@ public function forgotPasswordPostAction()
     {
         $email = (string) $this->getRequest()->getPost('email');
         if ($email) {
+            if (!Zend_Validate::is($email, 'EmailAddress')) {
+                $this->_getSession()->setForgottenEmail($email);
```

Comment:
One could argue that we also set the forgotten email back to empty string `""`, to make sure not break possibly break anything on the frontend. But I'll leave it as is for now.

Internally we've set it to empty string, just to make extra sure we dont pass in invalid data back to the frontend - even if its sesssion based and was user input.

Reply:
If I typed an invalid email, I would like to know what was submitted. That's just personal preference. But this change makes sense to me.

Reply:
An empty string wouldn't even pass the `if ($email) {` check

So I think your current code should be fine
but can you try to fix the PHP-CS-Fixer part?

It wants you to remove the extra spaces before the $flowpassword line

Reply:
It doesnt for the next request, thats true. Which is why the user must enter the email address again. If you pass in `setForgottenEmail($email);` the email address field on the frontend will be pre-filled with that value. This may or may not be wanted - for invalid email addresses.

Reply:
Agree not to set `setForgottenEmail('');`.